### PR TITLE
Cleanup binary version of clasp now that consumers have moved over

### PR DESCRIPTION
**Pull request by dibarbet**
Resolves https://github.com/dotnet/roslyn/issues/72251

PR val - https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/545199 - regressions in vmimages are known and unrelated.

### CODE REVIEW

File: src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/AbstractRequestScope.cs [hunk 1]
```diff
@@ -8,12 +8,7 @@
 using System;
 
 namespace Microsoft.CommonLanguageServerProtocol.Framework;
-
-#if BINARY_COMPAT // TODO - Remove with https://github.com/dotnet/roslyn/issues/72251
-public abstract class AbstractRequestScope(string name) : IDisposable
-#else
 internal abstract class AbstractRequestScope(string name) : IDisposable
```
Comment (ToddGrun) [inline src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/AbstractRequestScope.cs:11]: >internal [](http://example.com/codeflow?start=0&length=8)

extreme nit: newline

File: src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/IInitializeManager.cs [hunk 2]
```diff
@@ -6,12 +6,7 @@
 #nullable enable
 
 namespace Microsoft.CommonLanguageServerProtocol.Framework;
-
-#if BINARY_COMPAT // TODO - Remove with https://github.com/dotnet/roslyn/issues/72251
-public interface IInitializeManager<TRequest, TResponse>
-#else
 internal interface IInitializeManager<TRequest, TResponse>
```
Comment (ToddGrun) [inline src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/IInitializeManager.cs:9]: >internal [](http://example.com/codeflow?start=0&length=8)

nit: another missing newline

File: src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/ILspLogger.cs [hunk 3]
```diff
@@ -8,12 +8,7 @@
 using System;
 
 namespace Microsoft.CommonLanguageServerProtocol.Framework;
-
-#if BINARY_COMPAT // TODO - Remove with https://github.com/dotnet/roslyn/issues/72251
-public interface ILspLogger
-#else
 internal interface ILspLogger
```
Comment (ToddGrun) [inline src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/ILspLogger.cs:11]: >internal [](http://example.com/codeflow?start=0&length=8)

nit: another here, will stop marking these

**Comment by ToddGrun** [comment]
Love it!

**Review by ToddGrun — APPROVED** [review]
:shipit:

File: src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/ILspLogger.cs (hunk 3 above)
Comment (dibarbet) [reply src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/ILspLogger.cs:11]: I have one or two more followups in this area, so I'll defer these until then.

**Pull request merged**